src/eth/rpc/rpc_parser.rs

```diff
@@ -54,8 +60,28 @@ pub fn rpc_parsing_error<S: serde::Serialize>(message: S) -> ErrorObjectOwned {
     ErrorObjectOwned::owned(PARSE_ERROR_CODE, PARSE_ERROR_MSG, Some(message))
 }
 
-impl From<EthError> for ErrorObjectOwned {
-    fn from(error: EthError) -> Self {
-        rpc_internal_error(error.to_string())
+pub struct RPCError(anyhow::Error);
+impl Display for RPCError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        Display::fmt(&self.0, f)
+    }
+}
+impl Debug for RPCError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        Debug::fmt(&self, f)
     }
 }
+
+impl From<anyhow::Error> for RPCError {
```

Comment:
Check the [`gen_newtype_from`](https://github.com/cloudwalk/stratus/blob/93e35fa06d5497afb2156b357295b352173aae9e/src/ext.rs#L7) macro to derive `From` macros based on the inner type.

Example:
`gen_newtype_from!(self = RpcError, other = anyhow::Error, OtherErrorThatConvertToAnyhowError)`

Reply:
I'll take a look tomorrow.

Reply:
Not critical, but it just make things easier when we have this pattern to delegate to the wrapped type implementation.